File: finance-web/src/models/Category.ts
import type { CategoryType } from "../enum/CategoryType";

export interface CategoryDto {
    title: string;
    description: string;
}

export interface ReturnCategory {
    id: string;
    title: string;
    description: string;
    totalAmount: number;
    balance: number;
    income: number;
    expense: number;
    categoryType: CategoryType;
    createdAt: string;
}

export interface MinimalCategory {
    id: string;
    title: string;
    totalAmount: number;
    balance: number;
    categoryType: CategoryType;
    transactionsCount: number;
}

File: finance-api/Model/Category.cs
using finance_api.Middleware.Enum;

namespace finance_api.Model
{
    public class Category
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string Title { get; set; }
        public string Description { get; set; } = string.Empty;
        public decimal TotalAmount { get; set; } = 0;
        public decimal Balance {  get; set; } = 0;
        public decimal Income { get; set; } = 0;
        public decimal Expense { get; set; } = 0;
        public CategoryType CategoryType { get; set; }

        public DateTimeOffset CreateAt { get; set; } = DateTimeOffset.UtcNow;
        public bool isDeleted { get; set; } = false;

        public Guid UserId { get; set; }    
        public User User { get; set; } = null!;

        public void CategoryCondition()
        {
            if(Balance >= 0)
            {
                CategoryType = CategoryType.Profit;
            }
            else
            {
                CategoryType = CategoryType.Debt;
            }
        }
    }
}
